GTBitsOfGood/angels-among-us
# Add access

## Add Access

Issue Number(s): #7 

Adds functionality for adding accounts with the specified email/admin access. If there is a corresponding User with the same email, their access is updated and their account is enabled.

### Checklist


### Critical Changes

- None

### Related PRs

- None

### Testing

Test account creation:
- Run ```curl -X POST "http://localhost:3000/api/trpc/account.add" -H "Content-Type: application/json" -d '{"email": "EMAIL", "admin": TRUE/FALSE}'``` and check that the document shows up in MongoDB with the correct fields

Test User changes:
- Create a user with the same email and disabled key set to true
- Run the command with the admin field as the opposite of the user's current admin status
- Check that disabled gets set to false and admin gets changed

Test error conditions:
- Try to create an Account with the same email as an already existing Account
- Throw errors before/between/after calls to addAccount and updateUser and check that no changes occur in MongoDB
